frontend/src/hooks/useReceivingCaptions.tsx

```diff
@@ -0,0 +1,48 @@
+import { Subscriber } from '@vonage/client-sdk-video';
+import { useState, useEffect, useMemo } from 'react';
+
+export type CaptionsType = {
+  captionText: string;
+  isReceivingCaptions: boolean;
+};
+
+export type ReceivingCaptionsProps = {
+  subscriber?: Subscriber | null;
+};
+
+export type CaptionReceivedType = {
+  streamId: string;
+  caption: string;
+  isFinal: boolean;
+};
+
+/**
+ * Hook to manage receiving captions from a subscriber.
```

Comment:
Do we need this hook for a publisher subscribing to his own captions too?  If so, maybe rephrase as "from a speaker" instead?

Reply:
yes, we definitely do! great catch. this was added before I added the self-subscribe to captions. 🚀